packages/shared/forks/invokeGuardedCallbackImpl.www.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import invariant from 'shared/invariant';
+
+// Provided by www
+const ReactFbErrorUtils = require('ReactFbErrorUtils');
+invariant(
+  typeof ReactFbErrorUtils.invokeGuardedCallback === 'function',
+  'Expected ReactFbErrorUtils.invokeGuardedCallback to be a function.',
+);
+
+// This object will be mutated by the www version.
+let result = {
+  _hasCaughtError: false,
+  _caughtError: null,
+};
+
+let invokeGuardedCallbackImpl = function<A, B, C, D, E, F, Context>(
+  name: string | null,
+  func: (a: A, b: B, c: C, d: D, e: E, f: F) => mixed,
+  context: Context,
+  a: A,
+  b: B,
+  c: C,
+  d: D,
+  e: E,
+  f: F,
+) {
+  // The www version doesn't take the `onError` callback on context.
+  // Instead, it writes the result to fields on context.
+  // TODO: change the www function API so we can re-export it directly.
+  result._hasCaughtError = false;
+  result._caughtError = null;
```

Comment:
Technically www already resets these but it felt a bit fragile so I'm repeating them here.

Reply:
That's two extra, usually redundant, calls we have to do on every invocation. Is `invokeGuardedCallback` ever used in prod by www? If so we should get rid of these.

Reply:
It’s used in prod yeah. I can remove them. I don’t expect them to stay here for more than a few weeks (when I do the next sync and replace this code)

Reply:
The idea is that a call to `invokeGuardedCallback` must always be followed by a call to `hasCaughtError`, which does the reset. It's a low-level, infra-level API so the fragileness is preferable to the perf hit (unless it's DEV-only I suppose).

Reply:
Although if this were used in prod, we would probably mutate the object directly like ReactCurrentOwner, instead of calling a function.